### PR TITLE
Mm/correlation matrix

### PR DESCRIPTION
**Pull request by mm-binary**

### CODE REVIEW

File: lib/Quant/Framework/ExpiryConventions.pm [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+    return $expiry_date;
+}
+
+# When the term is expressed in months or years, the
+# logic for FX and EQ differs. Here is the FX logic.
+sub _FX_month_and_year_term_vol_expiry_date {
+    my ($self, $args)   = @_;
+    my ($from, $months) = @{$args}{qw(from months)};
+
+    # Step 1: To obtain the valid forward deliverty date
```
Comment (jy-binary) [inline lib/Quant/Framework/ExpiryConventions.pm]: typo in deliverty

File: lib/Quant/Framework/ExpiryConventions.pm [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+    }
+
+    # Step 3: To obtain the valid day_of_month for forward month
+
+# 3.1: First check if the spot date itself is drop on last business day of the spot month, then the delivery date should set as last day of forward month
+# Example: If the spot date is 31Jan2011, the 1M delivery date must set as 28-Feb-11 and not 31-Feb-11
+    my $first_day_of_next_mth_after_spot_mth = Date::Utility->new('1-' . $spot_date->months_ahead(1));
+    my $last_trading_day_of_spot_mth = $self->calendar->trade_date_before($first_day_of_next_mth_after_spot_mth, {lookback => 1});
+
+# This is just to build the BOM date for forward month so that we can obtain the last day of forward month
```
Comment (jy-binary) [inline lib/Quant/Framework/ExpiryConventions.pm]: rename "BOM date" - makes no sense in an open-source repo

File: lib/Quant/Framework/ExpiryConventions.pm (hunk 1 above)
Comment (mm-binary) [inline lib/Quant/Framework/ExpiryConventions.pm]: Fixed.

File: lib/Quant/Framework/ExpiryConventions.pm (hunk 2 above)
Comment (mm-binary) [inline lib/Quant/Framework/ExpiryConventions.pm]: Done.

File: t/correlation_matrix.t [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    my $expiry_conventions = Quant::Framework::ExpiryConventions->new(
+        chronicle_reader        => $chronicle_r,
+        for_date                => $date,
+        symbol                  => $index,
+        quoted_currency         => $qcurrency,
+        calendar                => $calendar,
+    );
+
+    my $tiy = 366 / 365;
```
Comment (shahrizal-shaari) [inline t/correlation_matrix.t]: I am just curious, why is 1 year = 366/365 ?

Comment (mm-binary) [inline t/correlation_matrix.t]: Obviously, It is not 1 year then. I changed the description.

**Comment by basiliscos** [comment]
Please, don't forget to maintain `Changes`

**Pull request merged**